heat/core/linalg/basics.py

```diff
@@ -155,6 +156,90 @@ def cross(
     return ret
 
 
+def det(a: DNDarray) -> DNDarray:
+    """
+    Returns the determinant of a square matrix.
+
+    Parameters
+    ----------
+    a : DNDarray
+        A square matrix or a stack of matrices. Shape = (...,M,M)
+
+    Raises
+    ------
+    TypeError
+        If the dtype of 'a' is not floating-point.
+    ValueError
+        If `a.ndim < 2` or if the length of the last two dimensions is not the same.
+
+    Examples
+    --------
+    >>> a = ht.array([[-2,-1,2],[2,1,4],[-3,3,-1]])
+    >>> ht.linalg.det(a)
+    DNDarray(54., dtype=ht.float64, device=cpu:0, split=None)
+    """
+    # no split in the square matrices
+    if not a.is_distributed() or a.split < a.ndim - 2:
+        data = torch.linalg.det(a.larray)
+        sp = None if not a.is_distributed() else a.split
+        return DNDarray(
+            data,
+            a.shape[:-2],
+            types.heat_type_of(data),
+            split=sp,
+            device=a.device,
+            comm=a.comm,
+            balanced=a.balanced,
+        )
+
+    if a.ndim < 2:
+        raise RuntimeError("DNDarray must be at least two-dimensional.")
```

Comment:
I'm not sure if we should follow the error types of PyTorch. It's RuntimeError mostly.